lib/agent.js

```diff
@@ -77,24 +81,30 @@ const createAgent = (exports.createAgent = (cookies = [], opts = {}) => {
     localAddress: options.localAddress,
     jar: options.cookies.jar,
   };
-});
+};
 
-exports.createProxyAgent = (options, cookies = []) => {
-  if (!cookies) cookies = [];
+const createProxyAgent = (options, cookies = []) => {
   if (typeof options === "string") options = { uri: options };
   if (options.factory) throw new Error("Cannot use factory with createProxyAgent");
+
   const jar = new CookieJar();
   addCookies(jar, cookies);
-  const proxyOptions = Object.assign(
-    {
-      factory: (origin, opts) => {
-        const o = Object.assign({ cookies: { jar } }, opts);
-        return new CookieClient(origin, o);
-      },
-    },
-    options,
-  );
+  
+  const proxyOptions = {
+    factory: (origin, opts) => new CookieClient(origin, { ...opts, cookies: { jar } }),
+    ...options,
+  };
+
   return { dispatcher: new ProxyAgent(proxyOptions), jar, localAddress: options.localAddress };
 };
 
-exports.defaultAgent = createAgent();
+// Default agent
+const defaultAgent = createAgent();
+
+module.exports = {
+  addCookies,
+  addCookiesFromString,
+  createAgent,
+  createProxyAgent,
+  defaultAgent,
+};
```

Comment:
Not necessary. But I will apply this if someone create a pull request contains this.